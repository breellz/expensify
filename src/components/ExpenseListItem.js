import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment'
import numeral from 'numeral'
import '../locales/nigeria'

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <Link className="list-item" to={`/edit/${id}`}>
      <div>
        <h3 className="list-item__title" >{description}</h3>
        <span  className="list-item__subtitle" >{moment(createdAt).format('Do MMM, YYYY')}</span>
      </div>
      <div>
      <h3 className="list-item__data" >{numeral(amount/ 100).format('$0,0.00')}</h3> 
      </div>
    </Link>
    
);

export default ExpenseListItem;
