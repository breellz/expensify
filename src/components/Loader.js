import React from 'react'

const Loader = () => (
    <div className="loader">
        <img className="loader__image"src='/images/loader.gif' alt="loader" />
    </div>
)
export default Loader