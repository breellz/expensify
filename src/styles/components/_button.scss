.button {
    background: $blue;
    color: white;
    font-weight: 300;
    display: inline-block;
    font-size: $font-size-large;
    padding: $s-size;
    border: none;
    text-decoration: none;
    line-height: 1;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}