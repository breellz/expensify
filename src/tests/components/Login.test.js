import React from 'react';
import { shallow } from 'enzyme';
import {Login} from '../../components/Login';

test('should render Login correctly', () => {
  const wrapper = shallow(<Login />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startlogin on button click', ()=>{
  const startLogin = jest.fn()
  const wrapper = shallow(<Login startLogin ={startLogin}/>)
  wrapper.find('button').simulate('click')
  expect(startLogin).toHaveBeenCalled()
})