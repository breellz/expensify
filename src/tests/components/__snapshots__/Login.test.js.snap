// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login correctly 1`] = `
<div>
  <button>
    Login!
  </button>
</div>
`;
